### PR TITLE
Add and remove List

### PR DESCRIPTION
**Pull request by karam084**
*- Project requirements

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.

### CODE REVIEW

File: src/modules/addTask.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  const listItemName = taskInput.value || 'New Item';
+  todoItems.listItem = createNewTaskElement(listItemName);
+  incompleteTasksHolder.appendChild(todoItems.listItem);
+  bindTaskEvents(todoItems.listItem, taskCompleted);
+  taskInput.value = '';
+  localStorage.setItem('addTask', JSON.stringify(listItemName));
+};
+if (JSON.parse(localStorage.getItem('listItemName')) != null) {
+  listItem.todoItems = JSON.parse(localStorage.getItem('listItemName'));
+}
```
Comment (keroloussamy) [inline src/modules/addTask.js]: - [ ] Kindly in your add method you should save the data in the local storage with the three properties `text, completed, index` for every task.
- [ ] Kindly you should when you try to reload the page you have to display the tasks that you have added, from the local storage.

image

File: src/modules/deleteTask.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const deleteTask = function () {
+  todoItems.listItem = this.parentNode;
+  const ul = todoItems.listItem.parentNode;
+  ul.removeChild(todoItems.listItem);
+  localStorage.setItem('deleteTask', JSON.stringify(listItem.todoItems));
+};
+if (JSON.parse(localStorage.getItem('todoItems')) != null) {
+  listItem.todoItems = JSON.parse(localStorage.getItem('deleteTask'));
+}
```
Comment (keroloussamy) [inline src/modules/deleteTask.js]: - [ ] Kindly in your delete method, you have to remove the task from the tasks array in your local storage, all your tasks will be in an array on the local storage and when you delete a task it should be removed from this array from local storage and from the page also.

image

File: src/taskInput.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+const editTask = newFunction();
+export const bindTaskEvents = (taskListItem, checkBoxEventHandler) => {
+  const checkBox = taskListItem.querySelector('input[type=checkbox]');
+  const editButton = taskListItem.querySelector('button.edit');
+  const deleteButton = taskListItem.querySelector('button.delete');
+  editButton.onclick = editTask;
+  deleteButton.onclick = deleteTask;
+  checkBox.onchange = checkBoxEventHandler;
+};
```
Comment (keroloussamy) [inline src/taskInput.js]: - [ ] Kindly in your edit method, when you edit a task it should be updated on the local storage also, so when you try to reload the page you find the updated data.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @karam084, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Check the comments under the review._

- [x] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
Hi @karam084 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**